classes/local/store/object_file_system.php

```diff
@@ -995,4 +995,40 @@ protected function get_trash_fullpath_from_hash($contenthash) {
         debugging('Objectfs does not implement a trashdir.');
         return '';
     }
+
+    /**
+     * Add the supplied file to the file system and update its location.
+     *
+     * @param string $pathname Path to file currently on disk
+     * @param string $contenthash SHA1 hash of content if known (performance only)
+     * @return array (contenthash, filesize, newfile)
+     */
+    public function add_file_from_path($pathname, $contenthash = null) {
+        $result = parent::add_file_from_path($pathname, $contenthash);
+
+        // Rather than getting its exact location we just set it to local.
+        // Almost all file uploads will be unique, and if it is a duplicate
+        // then this will be corrected when the file is synced later.
+        manager::update_object_by_hash($result[0], OBJECT_LOCATION_LOCAL, $result[1]);
```

Comment:
Could we do something like this:
```suggestion
        [$contenthash, $filesize] = parent::add_file_from_path($pathname, $contenthash);

        // Rather than getting its exact location we just set it to local.
        // Almost all file uploads will be unique, and if it is a duplicate
        // then this will be corrected when the file is synced later.
        manager::update_object_by_hash($contenthash, OBJECT_LOCATION_LOCAL, $filesize);
```

Based on the return array of add_file_from_path. Similar comment for the other function too.
Note the concern would be the duplicate contenthash variable name, where dual purpose usage is generally discouraged, looking at the implementation, it seems to update and return this value anyways.
Ref: https://github.com/moodle/moodle/blob/9ee4f8db8ba7f9c3596592a9e14c05c1ea5a024b/lib/filestorage/file_system_filedir.php

Reply:
Actually I saw the return value for this function, and I'm retracting my initial suggestion.